STUDIO/org.openl.rules.workspace/src/org/openl/rules/project/abstraction/RulesProject.java

```diff
@@ -594,4 +601,53 @@ public String getMainBusinessName() {
         }
         return folderPath.substring(folderPath.lastIndexOf('/') + 1);
     }
+    
+    private final Object tagsLock = new Object();
+    
+    public Map<String, String> getTags() {
+        synchronized (tagsLock) {
+            if (tags == null) {
+                tags = loadTags();
+            }
+            return tags;
+        }
+    }
+
+    private Map<String, String> loadTags() {
```

Comment:
You should use https://github.com/openl-tablets/openl-tablets/blob/4c6385312af20496c2d88d83877a3b758630d57e/DEV/org.openl.commons/src/org/openl/util/PropertiesUtils.java#L24 to read and write property files